Keas.Mvc/Models/ExpiringItemsViewModel.cs

```diff
@@ -55,9 +57,70 @@ public static async Task<ExpiringItemsViewModel> Create(ApplicationDbContext con
                 ExpiresBefore = expiresBefore,
                 ShowInactive = showInactive,
                 ItemList =  itemList,
-                ShowType = showType,                
+                ShowType = showType               
             };
             return viewModel;
         }
+
+        public static async Task<ReportItemsViewModel> CreateUnaccepted(ApplicationDbContext context, string teamName, bool showInactive, string showType, List<Role> userRoles)
+        {  
+            var expiringAccess = await context.AccessAssignments.IgnoreQueryFilters().Where(a => (showType == "All" || showType == "Access") && 
```

Comment:
I don't really like the idea here of adding queries to the iqueryable that don't have anything to do with the underlying db.  How does this get translated to SQL?  Are they just ignored or does it break up the SQL in some way?  I think overall if you just threw in a little private method which would determine if the info should show for a given type (i.e. "Workstation") it might simplify matters a lot and not be so repetitive. Also I'm wondering what happens from a SQL level if you don't have access to something?  Does the query short circuit before the SQL is generated based off the local arguments like the showType == "All"?

Reply:
Ignored:   The LINQ expression 'where ([r].Name == "DepartmentalAdmin")' could not be translated and will be evaluated locally.
Microsoft.EntityFrameworkCore.Query:Warning: The LINQ expression 'where ([r].Name == "DepartmentalAdmin")' could not be translated and will be evaluated locally.
warn: Microsoft.EntityFrameworkCore.Query[20500]
      The LINQ expression 'Any()' could not be translated and will be evaluated locally.
Microsoft.EntityFrameworkCore.Query:Warning: The LINQ expression 'Any()' could not be translated and will be evaluated locally.
warn: Microsoft.EntityFrameworkCore.Query[20500]
      The LINQ expression 'where ([r].Name == "AccessMaster")' could not be translated and will be evaluated locally.
Microsoft.EntityFrameworkCore.Query:Warning: The LINQ expression 'where ([r].Name == "AccessMaster")' could not be translated and will be evaluated locally.
warn: Microsoft.EntityFrameworkCore.Query[20500]
      The LINQ expression 'Any()' could not be translated and will be evaluated locally.
Microsoft.EntityFrameworkCore.Query:Warning: The LINQ expression 'Any()' could not be translated and will be evaluated locally.
warn: Microsoft.EntityFrameworkCore.Query[20500]
      The LINQ expression 'where ([r].Name == "Admin")' could not be translated and will be evaluated locally.
Microsoft.EntityFrameworkCore.Query:Warning: The LINQ expression 'where ([r].Name == "Admin")' could not be translated and will be evaluated locally.
warn: Microsoft.EntityFrameworkCore.Query[20500]
      The LINQ expression 'Any()' could not be translated and will be evaluated locally.
Microsoft.EntityFrameworkCore.Query:Warning: The LINQ expression 'Any()' could not be translated and will be evaluated locally.
warn: Microsoft.EntityFrameworkCore.Query[20500]
      The LINQ expression 'where (((True AndAlso (({from Role r in __userRoles_0 where ([r].Name == "DepartmentalAdmin") select [r] => Any()} OrElse {from Role r in __userRoles_1 where ([r].Name == "AccessMaster") select [r] => Any()}) OrElse {from Role r in __userRoles_2 where ([r].Name == "Admin") select [r] => Any()})) AndAlso ([a.Access.Team].Slug == __teamName_3)) AndAlso Not([a].IsConfirmed))' could not be translated and will be evaluated locally.
Microsoft.EntityFrameworkCore.Query:Warning: The LINQ expression 'where (((True AndAlso (({from Role r in __userRoles_0 where ([r].Name == "DepartmentalAdmin") select [r] => Any()} OrElse {from Role r in __userRoles_1 where ([r].Name == "AccessMaster") select [r] => Any()}) OrElse {from Role r in __userRoles_2 where ([r].Name == "Admin") select [r] => Any()})) AndAlso ([a.Access.Team].Slug == __teamName_3)) AndAlso Not([a].IsConfirmed))' could not be translated and will be evaluated locally.

Reply:
I'll experiment and see if the query fires off SQL even if the other bits are false.